modules/appeals_api/config/schemas/v2/legacy_appeals_headers.json

```diff
@@ -24,6 +24,13 @@
             { "$ref": "#/definitions/nonBlankString" },
             { "description": "Veteran's file number", "minLength": 1, "maxLength": 9 }
           ]
+        },
+        "X-VA-ICN": {
+          "type": "string",
+          "description": "Veteran's ICN",
+          "pattern": "^[0-9]{10}V[0-9]{6}$",
+          "minLength": 17,
+          "maxLength": 17
```

Comment:
Since the `X-VA-ICN` header is excluded from the headers to be validated when the feature flag is turned off, these header requirements can stay here.